server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java

```diff
@@ -0,0 +1,444 @@
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    // query settings
+    public static final Setting<TimeValue> CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_QUERY_WARN_SETTING = Setting.timeSetting(
+        CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX + ".query.warn",
+        TimeValue.timeValueNanos(-1),
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    public static final Setting<TimeValue> CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_QUERY_INFO_SETTING = Setting.timeSetting(
+        CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX + ".query.info",
+        TimeValue.timeValueNanos(-1),
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    public static final Setting<TimeValue> CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_QUERY_DEBUG_SETTING = Setting.timeSetting(
+        CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX + ".query.debug",
+        TimeValue.timeValueNanos(-1),
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    public static final Setting<TimeValue> CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_QUERY_TRACE_SETTING = Setting.timeSetting(
+        CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX + ".query.trace",
+        TimeValue.timeValueNanos(-1),
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    // fetch settings
+    public static final Setting<TimeValue> CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_FETCH_WARN_SETTING = Setting.timeSetting(
+        CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX + ".fetch.warn",
+        TimeValue.timeValueNanos(-1),
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    public static final Setting<TimeValue> CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_FETCH_INFO_SETTING = Setting.timeSetting(
+        CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX + ".fetch.info",
+        TimeValue.timeValueNanos(-1),
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    public static final Setting<TimeValue> CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_FETCH_DEBUG_SETTING = Setting.timeSetting(
+        CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX + ".fetch.debug",
+        TimeValue.timeValueNanos(-1),
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    public static final Setting<TimeValue> CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_FETCH_TRACE_SETTING = Setting.timeSetting(
+        CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX + ".fetch.trace",
+        TimeValue.timeValueNanos(-1),
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
```

Comment:
Does not make too much sense to have phase breakdown settings at the cluster level. Better will be to print the latency of different phases in the overall slow log message

Reply:
At shard it is necessary to have the phase breakdown settings as the correlation between query and fetch phase is lost. But, at the coordinator node we have the luxury of not losing that correlation

Reply:
Yes, unnecessary flexibility generally causes confusion. I want end user to have single setting that directly correlates with their SLAs. And log message from the overall setting should give the breakdown across different phases.

Why will user want to set threshold on individual query or fetch phase? As I said earlier, it was necessity at shard level due to missing correlation between the phases.

Reply:
Kind of agree with @jainankitk. Its more useful for users to set a request level threshold and have phase level breakdown as part of slowlogs.

Reply:
Done
1. Removed query and fetch phase logging thresholds
2. Added phase took time to logs (sample in PR description)